src/nwbinspector/tools/testing.py

```diff
@@ -0,0 +1,54 @@
+"""Helper functions for internal use in the test suite that rely on external dependencies (i.e., pynwb)."""
+from uuid import uuid4
+from datetime import datetime
+from typing import Optional, Tuple
+from urllib import request
+from shutil import rmtree
+from warnings import warn
+from tempfile import mkdtemp
+
+import h5py
+from pynwb import NWBFile
+from hdmf.testing import TestCase as HDMFTestCase
+
+
+def make_minimal_nwbfile():
+    """Most basic NWBFile that can exist."""
+    return NWBFile(session_description="", identifier=str(uuid4()), session_start_time=datetime.now().astimezone())
+
+
+def check_streaming_enabled() -> Tuple[bool, Optional[str]]:
+    """
+    General purpose helper for determining if the environment can support S3 DANDI streaming.
+
+    Returns the boolean status of the check and, if False, provides a string reason for the failure for the user to
+    utilize as they please (raise an error or warning with that message, print it, or ignore it).
+    """
+    try:
+        request.urlopen("https://dandiarchive.s3.amazonaws.com/ros3test.nwb", timeout=1)
+    except request.URLError:
+        return False, "Internet access to DANDI failed."
+    if "ros3" not in h5py.registered_drivers():
+        return False, "ROS3 driver not installed."
+    return True, None
+
+
+class TemporaryFolderTestCase(HDMFTestCase):
+    @classmethod
+    def setUpClass(cls):
+        """
+        The `tmpdir` pytest fixture does not immediately clean itself up after running the test suite.
+
+        Instead it caches some number of total testing runs under a temporary folder.
```

Comment:
Now, technically pytest will eventually clean up 'older' runs that it has cached - so it's mostly just personal preference that I think the testing suite should immediately and automatically clean itself up after each run for tests that write things to disk

You can verify this yourself by injecting something like

```python
def test_tmpdir(tmpdir):
    from pathlib import Path
    print(Path(tmpdir))  # To see save path on your system

    import numpy as np
    np.memmap(filename=Path(tmpdir)/"test.dat", shape=(10,2), dtype="int16", mode="w+")
    assert False # to force the print above to display in log
```

into a `test_<something>.py` file and running `pytest` directly on it several times, then watching what happens to the parent folder on the system